### PR TITLE
Fix issue where process.stdin is ended too early

### LINKED ISSUE #5320: Bun doesn't run prisma generate or prisma migrate inside docker containers or WSL
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

# Docker:
```dockerfile
FROM oven/bun AS builder

WORKDIR /tmp/app

COPY package.json .
COPY bun.lockb .

RUN bun install

COPY src ./src
COPY tsconfig.json .

# This would be the optimal solution but it doesn't work, so I'm gonna use a hack here until I figure out how to fix it
#   RUN bunx prisma:generate
#   COPY prisma/schema.prisma ./prisma/schema.prisma
COPY node_modules/.prisma ./node_modules/.prisma

RUN bun run build

FROM oven/bun AS runner

ARG DEFAULT_DATABASE_URL= "file:./deploy.db"
ENV DATABASE_URL=$DEFAULT_DATABASE_URL

WORKDIR /app

COPY --from=builder /tmp/app/package.json /app/package.json
COPY --from=builder /tmp/app/bun.lockb /app/bun.lockb
RUN bun install -p

# Refer to the comment above
#   COPY --from=builder /tmp/app/prisma/schema.prisma ./prisma/schema.prisma
#   RUN bun run prisma:generate
COPY --from=builder /tmp/app/node_modules/.prisma /app/node_modules/.prisma

#   These commands aren't working either. The prisma:migrate has to be done in the external machine
#   RUN bun add prisma
#   RUN bun run prisma:migrate

COPY --from=builder /tmp/app/build /app/build

CMD ["bun", "start"]
```

# WSL:

Install WSL normally...
```powershell
wsl --install
```

Make bun installation inside Ubuntu.

Try to follow this exact guide: https://bun.sh/guides/ecosystem/prisma

### What is the expected behavior?

Prisma installs normally

### What do you see instead?

This, download never finishes:
```bash
root@7fa00e64baa1:/app# bun run prisma:deploy
$ bunx prisma migrate deploy
Downloading Prisma engines for Node-API for debian-openssl-1.1.x [===                 ] 15%root@7fa00e64baa1:/app#
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by camero2734**
### What does this PR do?
Fixes #5320 (hopefully the last issue)

The issue this time is kind of similar to the last one (#9101), in that a stream is being ended too early. This one's a bit more straightforward, though I'm not sure if it's the correct behavior.

This issue only occurs with `process.stdin` since it has some custom logic in `src/js/builtins/ProcessObjectInternals.ts`. Turns out there's this line:
```ts
stream._read = function (size) {
  $debug("_read();", reader);
  if (!reader) {
    // TODO: this is wrong
    this.push(null);
  } else if (!shouldUnref) {
    internalRead(this);
  }
};
```

This TODO was, in fact, true. When a `null` is pushed, the stream ends, so this was causing the stream to be ended prematurely if `reader` wasn't defined (which happens if `ref()` isn't called before `_read()` is).

To fix this, I just made it do nothing. Once I did this, however, I also noticed that `end` is emitted twice, so I added a check for whether end was already emitted, similar to the standard code in `stream.js`.

### How did you verify your code works?
There's a test. You can also more easily test this manually by creating `test.js`:
```js
const { Transform } = require("node:stream");

let totalChunkSize = 0;
const uppercase = new Transform({
  transform(chunk, _encoding, callback) {
    totalChunkSize += chunk.length;
    callback(null, "");
  },
});

process.stdin.pipe(uppercase).pipe(process.stdout);
process.stdin.on("end", () => console.log(totalChunkSize));
```

and then run it with

```bash
printf 'a%.0s' {1..262164} | bun run test.js
```

It should output the correct size (262164). Previously it would output 196608.

I also tested with a large prisma schema -- it failed with the `JSON Parse error: Unterminated string` error before, but now successfully generates

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
nice

**Comment by Jarred-Sumner** [comment]
Can you add a test?

**Comment by camero2734** [comment]
Before:
image

After:
image

File: test/js/third_party/prisma/helper.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export function generate(type: string) {
 
   fs.writeFileSync(schema, content);
 
-  const result = Bun.spawnSync([bunExe(), "prisma", "generate", "--schema", schema], {
```
Comment (Jarred-Sumner) [inline test/js/third_party/prisma/helper.ts]: using `bun x` shouldn't be necessary here

File: test/js/third_party/prisma/helper.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export function generate(type: string) {
...
+  const result = Bun.spawnSync([bunExe(), "--bun", "x", "prisma", "generate", "--schema", schema], {
```
Comment (Jarred-Sumner) [inline test/js/third_party/prisma/helper.ts]:
```suggestion
  const result = Bun.spawnSync([bunExe(), "prisma", "generate", "--schema", schema], {
```

**Comment by Jarred-Sumner** [comment]
Thank you

**Comment by argosphil** [comment]
Can we close #8693 , then?

**Comment by argosphil** [comment]
Hmm. My fix in https://github.com/argosphil/bun/commit/38c60f62ba1b3dc3799caf9b8f9b9f5cc52da2be was a bit more involved, in that I called `internalRead` right away when a reader was added. Sorry again for forgetting about this, I'll do some more testing later to see whether that's still necessary.

File: test/js/third_party/prisma/helper.ts (hunk 1 above)
Comment (camero2734) [reply test/js/third_party/prisma/helper.ts]: Yeah, had meant to remove that, thanks!

**Pull request merged**